dtls/server-dtls-rw-threads.c

```diff
@@ -0,0 +1,422 @@
+        currTimeout = wolfSSL_dtls_get_current_timeout(ssl);
+        /* Set max file descriptor to be one more than one reading from. */
+        nfds = nb_sockfd + 1;
+        /* Set seconds of time out. */
+        timeout.tv_sec = (currTimeout > 0) ? currTimeout : 0;
+
+        /* Setup file descriptor list to monitor. */
+        FD_ZERO(&recvfds);
+        FD_SET(nb_sockfd, &recvfds);
+        FD_ZERO(&errfds);
+        FD_SET(nb_sockfd, &errfds);
+
+        /* Monitor file descriptors. */
+        result = select(nfds, &recvfds, NULL, &errfds, &timeout);
+        if ((result > 0) && FD_ISSET(nb_sockfd, &errfds)) {
+            /* Tell other threads to cleanup. */
+            args->cleanup = 1;
+            /* Finished reading. */
+            break;
+        }
+
+        /* If data waiting on receive file descriptor, read it. */
+        if (result > 0 && FD_ISSET(nb_sockfd, &recvfds)) {
+            /* Read application data. */
+            if ((recvLen = wolfSSL_read(ssl, buff, msgLen-1)) < 0) {
+                /* Handle errors. */
+                int readErr = wolfSSL_get_error(ssl, 0);
+                if (readErr != SSL_ERROR_WANT_READ) {
+                    /* Tell other threads to cleanup. */
+                    args->cleanup = 1;
+                    /* Finished reading. */
+                    break;
+                }
+            }
+        }
+    }
+
+    return NULL;
+}
+
+/* Writer thread.
+ *
+ * Only started after handshake complete.
+ */
+void* Writer(void* openSock)
+{
+    threadArgs* args = (threadArgs*)openSock;
+    char               msg[] = "I hear you fashizzle!\n";
+    unsigned int       msgLen = sizeof(msg);
+    WOLFSSL*           ssl = args->ssl;
+    int                len;
+
+    /* Keep writing while not in cleanup. */
+    while (!args->cleanup) {
+        /* Write message and check for error. */
+        if ((len = wolfSSL_write(ssl, msg, msgLen)) < 0) {
+            /* Tell other threads to cleanup. */
+            args->cleanup = 1;
+            /* Finished writing. */
+            break;
+        }
+    }
+
+    return NULL;
+}
+
+int main(int argc, char** argv)
+{
+    /* Loc short for "location" */
+    char          caCertLoc[] = "certs/ca-cert.pem";
+    char          servCertLoc[] = "certs/server-cert.pem";
+    char          servKeyLoc[] = "certs/server-key.pem";
+    WOLFSSL_CTX*  ctx;
+    /* Variables for awaiting datagram */
+    int           on = 1;
+    int           res = 1;
+    int           connfd = 0;
+    int           recvLen = 0;    /* length of message */
+    int           listenfd = 0;   /* Initialize our socket */
+    int           flags = fcntl(*(&listenfd), F_GETFL, 0);
+    WOLFSSL*      ssl = NULL;
+    socklen_t     cliLen;
+    socklen_t     len = sizeof(int);
+    unsigned char b[MSGLEN];      /* watch for incoming messages */
+    threadArgs    args;
+    pthread_t     threadidReader;
+    pthread_t     threadidWriter;
+    pthread_t     threadidEnc;
```

Comment:
```
server-dtls-rw-threads.c:242:19: warning: unused variable ‘threadidEnc’ [-Wunused-variable]
  242 |     pthread_t     threadidEnc;
      |                   ^~~~~~~~~~~
```
(note this var is indeed used when building with `-DWOLFSSL_THREADED_CRYPT`, but only then.)

Reply:
fixed